[ bug ] mismatched types

```sh
error[E0308]: mismatched types                                                                                                                                                                                
   --> /tmp/tmp.1vrLGmLkbB/work/rustic-f0ddb47/cargo-crates/rustic_core-0.9.0/src/backend/local_destination.rs:666:65                                                                                         
    |                                                                                                                                                                                                         
666 |                 mknod(&filename, SFlag::S_IFBLK, Mode::empty(), device)                                                                                                                                 
    |                 ----- arguments to this function are incorrect  ^^^^^^ expected `u64`, found `u32`                                                                                                      
    |                                                                                                                                                                                                         
note: function defined here                                                                                                                                                                                   
   --> /tmp/tmp.1vrLGmLkbB/work/rustic-f0ddb47/cargo-crates/nix-0.30.1/src/sys/stat.rs:155:8                                                                                                                  
    |                                                                                                                                                                                                         
155 | pub fn mknod<P: ?Sized + NixPath>(                                                                                                                                                                      
    |        ^^^^^                                                                                                                                                                                            
help: you can convert a `u32` to a `u64`                                                                                                                                                                      
    |                                                                                                                                                                                                         
666 |                 mknod(&filename, SFlag::S_IFBLK, Mode::empty(), device.into())                                                                                                                          
    |                                                                       +++++++                                                                                                                           

error[E0308]: mismatched types
   --> /tmp/tmp.1vrLGmLkbB/work/rustic-f0ddb47/cargo-crates/rustic_core-0.9.0/src/backend/local_destination.rs:692:65
    |
692 |                 mknod(&filename, SFlag::S_IFCHR, Mode::empty(), device)
    |                 ----- arguments to this function are incorrect  ^^^^^^ expected `u64`, found `u32`
    |
note: function defined here
   --> /tmp/tmp.1vrLGmLkbB/work/rustic-f0ddb47/cargo-crates/nix-0.30.1/src/sys/stat.rs:155:8
    |
155 | pub fn mknod<P: ?Sized + NixPath>(
    |        ^^^^^
help: you can convert a `u32` to a `u64`
    |
692 |                 mknod(&filename, SFlag::S_IFCHR, Mode::empty(), device.into())
    |                                                                       +++++++
```

commit: f0ddb47
cargo `1.93.0-nightly` 
OS: `FreeBSD 14`

--- a/crates/core/src/backend/local_destination.rs
+++ b/crates/core/src/backend/local_destination.rs
@@ -403,7 +403,7 @@ impl LocalDestination {
     /// * `item` - The item to set the permissions for
     /// * `node` - The node to get the permissions from
     ///
-    /// # Errors        
+    /// # Errors
     ///
     /// * If the permissions could not be set.
     #[allow(clippy::unused_self, clippy::unnecessary_wraps)]
@@ -423,7 +423,7 @@ impl LocalDestination {
     /// * `item` - The item to set the permissions for
     /// * `node` - The node to get the permissions from
     ///
-    /// # Errors        
+    /// # Errors
     ///
     /// * If the permissions could not be set.
     #[allow(clippy::similar_names)]
@@ -647,24 +647,12 @@ impl LocalDestination {
                 })?;
             }
             NodeType::Dev { device } => {
-                #[cfg(not(any(
-                    target_os = "macos",
-                    target_os = "openbsd",
-                    target_os = "freebsd"
-                )))]
+                #[cfg(not(any(target_os = "macos", target_os = "openbsd")))]
                 let device = *device;
                 #[cfg(any(target_os = "macos", target_os = "openbsd"))]
                 let device = i32::try_from(*device).map_err(|err| {
                     LocalDestinationErrorKind::DeviceIdConversionFailed {
                         target: "i32".to_string(),
-                        device: *device,
-                        source: err,
-                    }
-                })?;
-                #[cfg(target_os = "freebsd")]
-                let device = u32::try_from(*device).map_err(|err| {
-                    LocalDestinationErrorKind::DeviceIdConversionFailed {
-                        target: "u32".to_string(),
                         device: *device,
                         source: err,
                     }
@@ -673,24 +661,12 @@ impl LocalDestination {
                     .map_err(LocalDestinationErrorKind::FromErrnoError)?;
             }
             NodeType::Chardev { device } => {
-                #[cfg(not(any(
-                    target_os = "macos",
-                    target_os = "openbsd",
-                    target_os = "freebsd"
-                )))]
+                #[cfg(not(any(target_os = "macos", target_os = "openbsd")))]
                 let device = *device;
                 #[cfg(any(target_os = "macos", target_os = "openbsd"))]
                 let device = i32::try_from(*device).map_err(|err| {
                     LocalDestinationErrorKind::DeviceIdConversionFailed {
                         target: "i32".to_string(),
-                        device: *device,
-                        source: err,
-                    }
-                })?;
-                #[cfg(target_os = "freebsd")]
-                let device = u32::try_from(*device).map_err(|err| {
-                    LocalDestinationErrorKind::DeviceIdConversionFailed {
-                        target: "u32".to_string(),
                         device: *device,
                         source: err,
                     }

